src/liftdrag_plugin/liftdrag_plugin.cpp

```diff
@@ -271,28 +324,22 @@ void LiftDragPlugin::OnUpdate()
   double speedInLDPlane = velInLDPlane.Length();
   double q = 0.5 * this->rho * speedInLDPlane * speedInLDPlane;
 
+#if GAZEBO_MAJOR_VERSION >= 9
+    double dal = this->lAileronJoint->Position(0);
+    double dar = this->rAileronJoint->Position(0);
+    double de = this->elevatorJoint->Position(0);
+#else
+    double dal = this->lAileronJoint->GetAngle(0).Radian();
+    double dar = this->rAileronJoint->GetAngle(0).Radian();
+    double el = this->elevatorJoint->GetAngle(0).Radian();
+#endif
+
   // compute cl at cp, check for stall, correct for sweep
   double cl;
-  if (this->alpha > this->alphaStall)
-  {
-    cl = (this->cla * this->alphaStall +
-          this->claStall * (this->alpha - this->alphaStall))
-         * cosSweepAngle;
-    // make sure cl is still great than 0
-    cl = std::max(0.0, cl);
-  }
-  else if (this->alpha < -this->alphaStall)
-  {
-    cl = (-this->cla * this->alphaStall +
-          this->claStall * (this->alpha + this->alphaStall))
-         * cosSweepAngle;
-    // make sure cl is still less than 0
-    cl = std::min(0.0, cl);
-  }
-  else
-    cl = this->cla * this->alpha * cosSweepAngle;
+  cl = this->cla * this->alpha + this->cl0 + this->clda * (this->dal + this->dar) + this->clde * this->de;
```

Comment:
Take sideslip into account.

Reply:
do you know how it should be accounted for?

Reply:
For now we are effectively assuming Cl_beta is 0 AND sideslip velocity's conribution to dynamic pressure is negligible. I think this is a fine assumption to get started. I started an issue to remind us to look into this at some point